src/main/java/org/broadinstitute/hellbender/tools/spark/utils/HopscotchCollection.java

```diff
@@ -562,6 +557,25 @@ public T next() {
             final T result = buckets[currentIndex];
             removeIndex = currentIndex;
             removePrevIndex = prevIndex;
+            advanceUntilEquivalent();
+            return result;
+        }
+
+        @Override
+        public void remove() {
+            super.remove();
+            if ( currentIndex != NO_ELEMENT_INDEX ) {
+                checkEquivalence();
+            }
+        }
+
+        private void checkEquivalence() {
```

Comment:
I'm always a little wary about methods with names like `checkXXX` that actually have side effects, like in this case advancing the `currentIndex` pointer. Maybe something like `ensureEquivalence` would be better? Alternatively, could you just have `advanceUntilEquivalent` contain the initial check for equivalence and just use that?